### PR TITLE
[Feature] Support semantic seg metrics

### PR DESCRIPTION
**Pull request by junhaozhang98**

### CODE REVIEW

File: mmdet3d/core/evaluation/seg_eval.py [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+        minlength=num_classes**2)
+    return bin_count[:num_classes**2].reshape(num_classes, num_classes)
+
+
+def per_class_iou(hist):
+    """Compute the per class iou.
+
+    Args:
+        hist(np.ndarray):  Overall confusion martix
+        (num_classes, num_classes ).
```
Comment (Tai-Wang) [inline mmdet3d/core/evaluation/seg_eval.py:32]: Add a blank line before Returns.

File: mmdet3d/core/evaluation/seg_eval.py [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+
+    return np.diag(hist) / (hist.sum(1) + hist.sum(0) - np.diag(hist))
+
+
+def get_acc(hist):
+    """Compute the overall accuracy.
+
+    Args:
+        hist(np.ndarray):  Overall confusion martix
+        (num_classes, num_classes ).
```
Comment (Tai-Wang) [inline mmdet3d/core/evaluation/seg_eval.py:45]: Blank line

File: mmdet3d/core/evaluation/seg_eval.py [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+
+    return np.diag(hist).sum() / hist.sum()
+
+
+def get_acc_cls(hist):
+    """Compute the class average accuracy.
+
+    Args:
+        hist(np.ndarray):  Overall confusion martix
+        (num_classes, num_classes ).
```
Comment (Tai-Wang) [inline mmdet3d/core/evaluation/seg_eval.py:58]: Blank line.

File: mmdet3d/core/evaluation/seg_eval.py [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+    return np.nanmean(np.diag(hist) / hist.sum(axis=1))
+
+
+def seg_eval(gt_labels, seg_preds, label2cat, logger=None):
+    """Semantic Segmentation  Evaluation.
+
+     Evaluate the result of the Semantic Segmentation.
+
+     Args:
+         gt_labels (list[torch. Tensor]): Ground truth labels.
```
Comment (Tai-Wang) [inline mmdet3d/core/evaluation/seg_eval.py:72]: torch.Tensor

File: mmdet3d/core/evaluation/seg_eval.py [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+    miou = np.nanmean(iou)
+    acc = get_acc(sum(hist_list))
+    acc_cls = get_acc_cls(sum(hist_list))
+
+    header = ['classes']
+    for i in range(len(label2cat)):
+        header.append(label2cat[i])
+    header.append('miou')
+    header.append('acc')
+    header.append('acc_cls')
```
Comment (Tai-Wang) [inline mmdet3d/core/evaluation/seg_eval.py:98]: Merge these appends?

Comment (junhaozhang98) [reply mmdet3d/core/evaluation/seg_eval.py:101]: Maybe only can append by 'for' or this way.

Comment (Tai-Wang) [reply mmdet3d/core/evaluation/seg_eval.py:101]: Actually you can set a list at first and use `.extend`. That will be clearer and more concise.

**Review by Tai-Wang — APPROVED** [review]

File: mmdet3d/core/evaluation/seg_eval.py (hunk 5 above)
Comment (junhaozhang98) [reply mmdet3d/core/evaluation/seg_eval.py]: Got it. I have changed it. Thanks.

**Review by ZwwWayne — APPROVED** [review]

**Pull request merged**